.github/workflows/spellcheck.yml

```diff
@@ -0,0 +1,25 @@
+name: Documentation Checks
```

Comment:
🤔 Why do we need a separate build for this? Why not make the `SpellCheck` target part of the normal build?

Reply:
Now, the build.yml ignores changes inside the docs directory. If you are good to run a whole build process when only updating the docs, I will merge this to files together.

Reply:
I think I like having it as a separate build (but I'm no expert in all the pros and cons here)

Benefits of having it a separate build
* no need to run spellcheck for all three operating systems
* no need to run spell check when not modifying docs
* no need to run regular unit-tests when only modifying docs

Reply:
The biggest con is that you will only get the spell checking issues after your PR is build. On the other hand, I guess most people won't run the local build script anyway. So maybe keeping it separate is fine after all. 

And maybe we can invoke it as part of the `run.bat` in the `docs` folder?

Reply:
Sounds good. 

Before or after running jekyll? Or does this matter in any way?

Reply:
> What if we add a note in the contributing guidelines to run `build.(sh|ps1) --target spellcheck` before pushing (if the docs are effected)

We should most definitely do that. 

> Before or after running jekyll? Or does this matter in any way?

I guess doing this before building the site makes more sense. And `run.bat` is blocking.

Reply:
Ok.. added both :)